### PR TITLE
Adds Jamjar Glasses

### PR DESCRIPTION
**Pull request by AnonymousNow**
Now's PR experiments #3: Adding an item to a vending machine. Part of a series of tests for me to figure out how the code works a little better.

These are Jamjar Glasses, so called because they're as big and thick as the bottom of jam jars. They're a comical "fashion" item alternative to regular prescription glasses, with all the same functionality. They appear in the Clothesmate vending machine, where they replace one of two existing pairs of prescription glasses.
glasses

:cl: 
add: Nerd Co. has sent pairs of thicker prescription glasses out to Nanotrasen stations, for your local geek to wear. 
/:cl:

### CODE REVIEW

File: code/modules/clothing/glasses/glasses.dm [hunk 1]
```diff
@@ -136,6 +136,13 @@
 	item_state = "glasses"
 	vision_correction = 1 //corrects nearsightedness
 
+/obj/item/clothing/glasses/regular/jamjar
+	name = "Jamjar Glasses"
+	desc = "Also known as Virginity Protectors."
+	icon_state = "jamjar_glasses"
+	item_state = "jamjar_glasses"
+	vision_correction = 1 //corrects nearsightedness
```
Comment (ChangelingRain) [inline code/modules/clothing/glasses/glasses.dm]: this is a subtype so you don't need this line

Comment (AnonymousNow) [inline code/modules/clothing/glasses/glasses.dm]: How does that work? Its position in the file after the original prescription glasses, or something I'm missing? Everything I've done so far is based off existing code, so I'm a little afraid to deviate.

Comment (ShadowDeath6) [inline code/modules/clothing/glasses/glasses.dm]: glasses/regular has vision_correction set to 1, so anything with the path including that will also have vision_correction set to 1. look at https://github.com/tgstation/tgstation/pull/24373/files#diff-d7594f2da0715636602b06ec080ea828R146 , notice how it's still perscription glasses even though it does have a var for vision correction

Comment (AnonymousNow) [inline code/modules/clothing/glasses/glasses.dm]: So everything downstream in a path like that has those effects? Good to know. I think that sheds some light on what's going wrong with my next experiment. Thanks.

**Comment by PKPenguin321** [comment]
Malformed changelog. Preface the line describing your addition with "add: "

Also, superaggrocrag would love these for Hedwig Algebra.

File: code/modules/clothing/glasses/glasses.dm (hunk 1 above)
Comment (RemieRichards) [inline code/modules/clothing/glasses/glasses.dm]: @AnonymousNow your type is a CHILD of it's PARENT
```
child: /obj/item/clothing/glasses/regular/jamjar
parent: /obj/item/clothing/glasses/regular
```
and it INHERITs from it's PARENT all of it's PARENT's variables.

**Comment by FantasticFwoosh** [comment]
Now we need to give a purple pair to the RD 

Y'know for seeing moonlight(ing) in UV light

**Comment by Supermichael777** [comment]
ew fucking animu bullshit

**Comment by ShadowDeath6** [comment]
Grr anime as I support prs that add anime meme patting :^)

**Comment by Repukan** [comment]
Wasn't there something the citrus said about no character references

**Comment by ShadowDeath6** [comment]
these glasses are more of a trope in shows rather than a reference to a character

**Comment by Repukan** [comment]
That why they look exactly the same as their profile picture and character art?

**Comment by Repukan** [comment]
If you're talking about the anime glasses I think you're talking about, they have a weird hypnosis ellipse that ends in the middle of the glasses.

**Comment by AnonymousNow** [comment]
Strange - I don't watch anime, and the thought of having thick nerd glasses rose in what I thought to be an unbidden way from my brain (though characters like Bottles from Banjo-Kazooie come to mind, as do some people I know IRL with glasses this thick). I was originally going to add a swirl, but I decided that I preferred how the circles look.

Does anime invade your mind?

**Comment by Repukan** [comment]
Don't project.

A quick Google search into jamjar glasses brought up cosplay stuff, and the only trope I could find with glasses and cosplay was something about swirly glasses and old otakus.

If you mean about glasses thick as bottle bottoms or jars, the ones I can think off the top of my mind would be from Harry Potter. But those were clear without the weird double circle these have.

These seem a lot like what SAC has in his profile picture, hence my saying it seems like a player reference.

**Comment by AnonymousNow** [comment]
I should clarify - I was asking if anime can invade one's mind. As in, is it quite possible that incidental anime exposure (ie. being on 4chan, where anime comes up even if it's nothing to do with the topic, usually reaction images) led me to the idea of making swirly glasses? I wasn't suggesting that you were anime-obsessed; that was a bit of ambiguous typing on my part. I apologise.

Does Harry Potter have glasses that thick? Maybe on the cover of the books, but not in the films, at least. I was thinking of things like Bottom, or that guy from The Trailer Park Boys, or the VI unit way back in school. But looking at that picture, they do look alike; colour's off, though. It's possible that they influenced me as well.

Google "jamjar bottom glasses" to get the idea. Maybe it's more of a British thing.

**Comment by Repukan** [comment]
I mean, I understand the glasses you describe, Bubbles from trailer park boys came to my mind as well.

But those glasses are clear, without the dual circles. If you wanted to imitate the trope, then it would be more fitting to remove said inner circles, else it looks too much like SAC avatar glasses, which pushes it into player reference

**Comment by AnonymousNow** [comment]
Bubbles, that's his name? I was thinking of glasses in the same category, but I was remembering the glasses from the VI unit of my youth, which really did have glasses with circles like that (they were a little closer together and more slender, but that's hard to depict on a sprite this small). Again, like the ones @ Google "Jamjar Bottle Glasses".

Do they really look so much like SAC's sprite? I'm quite attached to them and I'm not sure how I'd change them without altering the shape, which is quite iconic of nerd glasses, IMO.

**Comment by Repukan** [comment]
Yup, they look too much alike.

That's just my opinion, ask oranges or the actual coders or the people that control that area tbh

**Comment by AnonymousNow** [comment]
Fair enough. I'll leave it up, then, and see what people think. A merger would know better than me on this point.

**Comment by PKPenguin321** [comment]
When I see them I think of that professor guy from luigis mansion

**Comment by AnonymousNow** [comment]
Elvin Gadd! I'd forgotten about him.

**Pull request merged**